src/qt/window.cpp

```diff
@@ -350,6 +350,9 @@ void wxWindowQt::PostCreation(bool generic)
     wxWindowBase::SetBackgroundColour(wxColour(GetHandle()->palette().background().color()));
     wxWindowBase::SetForegroundColour(wxColour(GetHandle()->palette().foreground().color()));
 
+    m_qtWindow->setMinimumSize(QSize(std::max(m_minWidth, 0), std::max(m_minHeight, 0)));
+    m_qtWindow->setMaximumSize(QSize(m_maxWidth == -1 ? QWIDGETSIZE_MAX : m_maxWidth, m_maxHeight == -1 ? QWIDGETSIZE_MAX : m_maxHeight));
```

Comment:
I believe there are a few issues here:
- there is a `wxMax()` that should probably be used;
- IIRC `-1` are special values for coordinates, meaning that any coordinate that is passed `-1` should keep its value, and `0` should reset it to the min / max possible value, in which case this and the other functions that follow need significant changes.